packages/grafana-ui/src/components/TimePicker/TimeRangePicker/TimeRangeForm.tsx

```diff
@@ -74,7 +74,7 @@ export const TimeRangeForm: React.FC<Props> = props => {
 
       props.onApply(timeRange);
     },
-    [from, to, roundup, timeZone]
+    [from.invalid, from.value, props, timeZone, to.invalid, to.value]
```

Comment:
nit: both timeZone and props seems strange. Maybe destruct props and use the props used in dependencies.

Reply:
agree with @hugohaggmark but as a minimum we should replace `props` with `props.onApply`.

Reply:
done